app/src/org/commcare/fragments/TaskConnectorFragment.java

```diff
@@ -67,6 +68,22 @@ public void cancelTask() {
         }
     }
 
+    /**
+     * @return Task is in a state where, if cancelled, it can dismiss its
+     * blocking dialog and allow the user them to continue using the app
+     */
+    public boolean canDetachFromCancelledTask() {
+        return currentTask == null || currentTask.canStopUIBlockOnCancel();
+    }
+
+    public boolean isDialogCancelButtonEnabled() {
+        return isTaskDialogCancelButtonEnabled;

```

Comment:
this can come straight from the task itself (even as a direct, synchronous method call), and there's no need to double-store the state in a new boolean